CoreScriptsRoot/ServerCoreScripts/ClientChat/ChannelsTab.lua

```diff
@@ -0,0 +1,251 @@
+
+	end
+end
+
+function methods:RenderDisplayText()
+	
+end
+
+function methods:FadeOutBackground(duration)
+	self.BackgroundTweener:Tween(duration, 1)
+end
+
+function methods:FadeInBackground(duration)
+	self.BackgroundTweener:Tween(duration, 0)
+end
+
+function methods:FadeOutText(duration)
+	self.TextTweener:Tween(duration, 1)
+end
+
+function methods:FadeInText(duration)
+	self.TextTweener:Tween(duration, 0)
+end
+
+function methods:CreateTweeners()
+	self.BackgroundTweener:CancelTween()
+	self.TextTweener:CancelTween()
+
+	self.BackgroundTweener = moduleTransparencyTweener.new()
+	self.TextTweener = moduleTransparencyTweener.new()
+
+	--// Register BackgroundTweener objects and properties
+	self.BackgroundTweener:RegisterTweenObjectProperty(self.UnselectedFrame, "BackgroundTransparency")
+	self.BackgroundTweener:RegisterTweenObjectProperty(self.SelectedFrame.BackgroundImage, "BackgroundTransparency")
+	self.BackgroundTweener:RegisterTweenObjectProperty(self.SelectedFrame.BlueBarLeft, "ImageTransparency")
+	self.BackgroundTweener:RegisterTweenObjectProperty(self.SelectedFrame.BlueBarRight, "ImageTransparency")
+
+	--// Register TextTweener objects and properties
+	self.TextTweener:RegisterTweenObjectProperty(self.NameTag, "TextTransparency")
+	self.TextTweener:RegisterTweenObjectProperty(self.NameTag, "TextStrokeTransparency")
+	self.TextTweener:RegisterTweenObjectProperty(self.NewMessageIcon, "ImageTransparency")
+	self.TextTweener:RegisterTweenObjectProperty(self.WhiteTextNewMessageNotification, "TextTransparency")
+	self.TextTweener:RegisterTweenObjectProperty(self.WhiteTextNewMessageNotification, "TextStrokeTransparency")
+
+	--print("Dumping:", self:Dump(), "||", self.BackgroundTweener:Dump())
+	--print("Dumping:", self:Dump(), "||", self.TextTweener:Dump())
+end
+
+--///////////////////////// Constructors
+--//////////////////////////////////////
+ClassMaker.RegisterClassType("ChannelsTab", methods)
+
+function module.new(channelName)
+	local obj = {}
+
+	local BaseFrame, NameTag, NewMessageIcon, UnselectedFrame, SelectedFrame = CreateGuiObject()
+	obj.GuiObject = BaseFrame
+	obj.NameTag = NameTag
+	obj.NewMessageIcon = NewMessageIcon
+	obj.UnselectedFrame = UnselectedFrame
+	obj.SelectedFrame = SelectedFrame
+
+	--// These four aren't used, but need to be kept as 
+	--// references so they wont be garbage collected in 
+	--// the tweener objects until this Tab object is 
+	--// garbage collected when it is no longer in use.
+	obj.BlueBarLeft = SelectedFrame.BlueBarLeft
+	obj.BlueBarRight = SelectedFrame.BlueBarRight
+	obj.BackgroundImage = SelectedFrame.BackgroundImage
+	obj.WhiteTextNewMessageNotification = obj.NewMessageIcon.TextLabel
+
+	obj.ChannelName = channelName
+	obj.UnreadMessageCount = 0
+	obj.Active = false
+
+	obj.BackgroundTweener = moduleTransparencyTweener.new()
+	obj.TextTweener = moduleTransparencyTweener.new()
+
+	obj.GuiObject.Name = "Frame_" .. obj.ChannelName
+
+	local maxNameLenght = 12
```

Comment:
I am aware of this text clipping for long names issue as in it doesn't work perfectly. It's a coming soon (tm) thing for second round improvements.